ddtrace/_trace/trace_handlers.py

```diff
@@ -153,6 +154,43 @@ def _maybe_start_http_response_span(ctx: core.ExecutionContext) -> None:
         )
 
 
+def _use_html(headers) -> bool:
+    """decide if the response should be html or json.
+
+    Add support for quality values in the Accept header.
+    """
+    ctype = headers.get("Accept", headers.get("accept", ""))
+    if not ctype:
+        return False
+    html_score = 0.0
+    json_score = 0.0
+    ctypes = ctype.split(",")
+    for ct in ctypes:
+        if len(ct) > 128:
+            # ignore long (and probably malicious) headers to avoid performances issues
+            continue
+        m = re.match(r"([^/;]+/[^/;]+)(?:;q=([01](?:\.\d*)?))?", ct.strip())
+        if m:
+            if m.group(1) == "text/html":
+                html_score = max(html_score, min(1.0, float(1.0 if m.group(2) is None else m.group(2))))
```

Comment:
What does the value of `m.group(2)` actually represent?

Reply:
group(0) is the entire match, group(1) is the first capturing subgroup `([^/;]+/[^/;]+)` that will be the mime type and group(2) is the second capturing subgroup `([01](?:\.\d*)?))` which should be a float between 0 and 1. As the second group is optional (inside a (...)?), it can be None even if a match happens.